### PR TITLE
GKE enhancement of some complex options

### PR DESCRIPTION
**Pull request by orangedeng**
Mirror PR for https://github.com/rancher/rancher/pull/17237

Problem:
We should expose some extra parameters for options, because some customers need them.

Solved:
GKE enhancement to add more complex options

Issues: 

https://github.com/rancher/rancher/issues/16411
https://github.com/rancher/rancher/issues/10922
https://github.com/rancher/rancher/issues/13980
https://github.com/rancher/rancher/issues/14298
https://github.com/rancher/rancher/issues/14634

UI Issues:

https://github.com/rancher/rancher/issues/16412
https://github.com/rancher/rancher/issues/17990
https://github.com/rancher/rancher/issues/17991
https://github.com/rancher/rancher/issues/17992
https://github.com/rancher/rancher/issues/17993

Related PR:
https://github.com/rancher/kontainer-engine/pull/106

### CODE REVIEW

**Comment by orangedeng** [comment]
@loganhz @cjellick

File: server/capabilities/gke_capabilities.go [hunk 1]
```diff
@@ -4,21 +4,26 @@ import (
...
 	"os"
 	"strings"
+
+	"github.com/rancher/kontainer-engine/drivers/gke"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	computev1 "google.golang.org/api/compute/v1"
+	containerv1 "google.golang.org/api/container/v1"
+	iamv1 "google.golang.org/api/iam/v1"
```
Comment (nathan-jenan-rancher) [inline server/capabilities/gke_capabilities.go]: Can we let these package names default? I don't generally like overriding the default names unless there is a collision.  _Especially_ so if we are indicating a version in our package names.  That may be confusing down the line if we update to v2 and someone forgets to update the name.

**Comment by orangedeng** [comment]
Updated

**Comment by nathan-jenan-rancher** [comment]
LGTM

**Pull request merged**